src/module-elasticsuite-catalog-optimizer/Ui/DataProvider/AddSearchContainerFieldToCollection.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Smile ElasticSuite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteCatalogOptimizer
+ * @author    Pierre Gauthier <pierre.gauthier@smile.fr>
+ * @copyright 2021 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+namespace Smile\ElasticsuiteCatalogOptimizer\Ui\DataProvider;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Data\Collection;
+use Magento\Ui\DataProvider\AddFieldToCollectionInterface;
+
+/**
+ * Add request type field to collection
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteCatalogOptimizer
+ * @author   Pierre Gauthier <pierre.gauthier@smile.fr>
+ */
+class AddSearchContainerFieldToCollection implements AddFieldToCollectionInterface
+{
+    /** @var ResourceConnection */
+    private $resource;
+
+    /**
+     * AddSearchContainerFieldToCollection constructor.
+     * @param ResourceConnection $resource
+     */
+    public function __construct(ResourceConnection $resource)
+    {
+        $this->resource = $resource;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addField(Collection $collection, $field, $alias = null)
+    {
+        $collection->join(
+            ['search_container' => $this->resource->getTableName('smile_elasticsuite_optimizer_search_container')],
+            'search_container.optimizer_id= main_table.optimizer_id',
+            [$field]
+        );
```

Comment:
Actually, a straight join will not work for optimisers configured with multiple request types, the relation between smile_elasticsuite_optimizer and smile_elasticsuite_optimizer_search_container is 1..N

You will get duplicate smile_elasticsuite_optimizer rows the the classical 
```
Item (Smile\ElasticsuiteCatalogOptimizer\Model\Optimizer) with the same ID "1" already exists.
```